src/dmd/backend/iasm.d

```diff
@@ -499,7 +499,7 @@ union PTRNTAB {
 
 struct OP
 {
-    const(char)* str;   // opcode string
+    string str;   // opcode string
```

Comment:
I don't know if the header generator generates the proper struct for D arrays, but I can make it a union or use a `version` if this interferes with DMC. This is needed for CTFE to ease comparison of strings at compile-time.

Reply:
Header generator for what? DMC is written in D if you have C++ interop in mind

Reply:
> Header generator for what? DMC is written in D if you have C++ interop in mind

I don't know how walter interfaces this backend with DMC (https://github.com/DigitalMars/dmc). I don't think it is made in D but if this is a really old tree, then the change doesn't matter. If it does, I assumed the interfaces are not written by hand.

Reply:
Right, so nevermind. I guess all the extern(C++) thing and __gshared can go away then?

Reply:
Yes.